src/config_file.c

```diff
@@ -130,6 +134,12 @@ gboolean load_config(const gchar *filename, RaucConfig **config, GError **error)
 		c->system_bb_statename = g_key_file_get_string(key_file, "system", "barebox-statename", NULL);
 	}
 
+	c->max_download_size = g_key_file_get_uint64(key_file, "system", "max-download-size", NULL);
+	if (!c->max_download_size) {
```

Comment:
`g_key_file_get_uint64()` distinguishes among others between `G_KEY_FILE_ERROR_KEY_NOT_FOUND` and `G_KEY_FILE_ERROR_INVALID_VALUE`, so please utilize the function's error argument instead of its return value and apply a finer grained differentiation with different debug messages, but the same default value (see [here](https://github.com/rauc/rauc/blob/master/src/config_file.c#L456) and [here](https://github.com/rauc/rauc/blob/master/src/install.c#L967) for example).